djangobulk/bulk.py

```diff
@@ -139,7 +140,7 @@ def _update_many(model, objects, keys=None, using="default", skip_result=True,
     key_fields = [f for f in model._meta.fields if f.name in keys]
     value_fields = [
         field for field in _model_fields(model, update_fields)
-        if field.name not in keys
+        if field.name not in (keys + exclude_fields)
```

Comment:
This now is problematic if keys is a `list` and `exclude_fields`, by default it a tuple:

```
    if field.name not in (keys + exclude_fields)
TypeError: can only concatenate tuple (not "list") to tuple
```

Reply:
@fathineos ^^

Reply:
@diegobz I don't see how `exclude_fields` are a tuple by default, but I understand your concern about concatenating list and tuples. The only way this could happen is if the client passes a tuple to `keys` or `exclude_fields`, but again both in `update_many` and `insert_or_update_many`, `keys` and `exclude_fields` are documented to be lists. So do you think `django-bulk` should safeguard from this or should we leave it as is?

Also I will fix the typos

Reply:
> This now is problematic if keys is a list and exclude_fields, by default it a tuple
I meant the other way around indeed. :)
If the `keys` is a tuple and `exclude_fields` being by default a list.

I think it's a good idea to make sure both objects are of the same type before concatenating them. This update will break people's code if they are passing `keys` as a tuple inadvertently, which indeed used to work anyways. This is how I came across it.

And looking at our main codebase, it seems indeed we have tuples being passed.

Reply:
@diegobz a) this is a public repo (not an internal transifex repository), b) on principle, undocumented behaviour should not be relied on, therefore code that passes tuples to this method was always problematic and should be fixed.

Having said the above, I think there are a few reasons to change this piece of code:
1. it can be more useful if it accepts any iterable (as long as that is documented)
2. I feel this is probably a waste of cpu resources, doing the concatenation once for every  mode field.

I'll make a PR that fixes these issues.

Reply:
@igavriil @diegobz https://github.com/transifex/django-bulk/pull/9